test/whitequark/test_it_param_precedence_0.rb

```diff
@@ -0,0 +1,9 @@
+# typed: true
+
+# Test precedence: local variable 'it' takes precedence over block parameter
+it = "string"
+result = [1, 2, 3].map { it.upcase } # Should use the local variable 'it' which is "string"
+# Result is ["STRING", "STRING", "STRING"]
```

Comment:
Could you add a `T.reveal_type(result)` here, and assert that this produces an array of strings?

Reply:
I think this is a copy/past error. Integers don't have an `upcase`.

Anyway, for asserting the type I'd recommend:

```
T.assert_type!(result, T::Array[String])
```

Reply:
We definitely prefer `T.reveal_type` to `T.assert_type!` in testdata tests.

In this case, if something were wrong and the `it` logic changed such that it was calling a method on the elements of the array, then the result type of the `.map` would be `T::Array[T.untyped]` (arising from the error recovery on the errant method call), which would still have type `T::Array[String]` according to the `T.assert_type!` but would fail the test if using `T.reveal_type`.